execution_engine/src/storage/error/db.rs

```diff
@@ -14,6 +15,19 @@ pub enum Error {
     #[error(transparent)]
     Lmdb(#[from] lmdb_external::Error),
 
+    /// RocksDb error
+    #[error("{0}")]
+    RocksDb(String),
```

Comment:
```suggestion
    #[error(transparent)]
    RocksDb(#[from] rocksdb::Error),
```
(and you may need to remove derive Eq)

As historically storage in EE had leaky abstractions I'd appreciate it if we'd plan to have a clean abstraction starting with errors in `storage::Error`. Have a single variant `Database(Box<dyn std::error::Error>)` that contains either lmdb or rocksdb error at a minimum. Or a few generic error categories represented as variants that will be translated from DB-specific errors similar to https://github.com/casper-network/casper-node/blob/d0b3e00a862631b249a125e4bc2227041b132c99/node/src/components/storage/lmdb_ext.rs#L26-L41

Reply:
Ok I've decided to break out a separate DbError enum that explicitly handles lmdb::Error and rocksdb::Error as well as our own errors here. Newtyping over String here was done as a stopgap.